src/box/gc.c

```diff
@@ -674,7 +684,9 @@ gc_consumer_register(const struct vclock *vclock, const char *name, const struct
 	consumer->uuid = *uuid;
 
 	vclock_copy(&consumer->vclock, vclock);
-	gc_tree_insert(&gc.active_consumers, consumer);
+	consumer->is_inactive = gc_consumer_is_outdated(consumer);
+	if (!consumer->is_inactive)
+		gc_tree_insert(&gc.active_consumers, consumer);
```

Comment:
Fixed, left `is_inactive` semantics intact.

> If a consumer becomes inactive, it should be deleted from the _gc_consumers system space.

Since design of this patch mostly follows KISS principle, let's not delete them. Anyway, deactivation is extremely rare, and when consumer of replica is deactivated, it will soon fail with `XlogGapError` and will be deleted from the cluster - that's when the inactive consumer will be dropped. Otherwise, the feature needs to be thoroughly tested by an error injection.